tasks/system_probe.py

```diff
@@ -256,7 +256,7 @@ def build_object_files(ctx, install=True):
 
     for d in linux_headers:
         for s in subdirs:
-            flags.extend(["-I", os.path.join(d, s)])
+            flags.extend(["-isystem", os.path.join(d, s)])
```

Comment:
why does this need to change?

Reply:
This is indeed somewhat unrelated.
This allows to remove all the `#pragma clang diagnostic ignored …` that there are around `#include <linux/ptrace.h>` and `#include <net/sock.h>` in `pkg/ebpf/c/tracer-ebpf.c`.
Indeed, those two included files are system headers and they may generate compiler warnings, depending on the warnings that we have enabled few lines above in the same file.
I understand that we want aggressive warning checks for our own code (`tracer-ebpf.c`) but we won’t fix the system headers to make them compliant with our own warning options.
That’s why there used to be some `#pragma clang diagnostic ignored …` lines.
But when I moved to a more up-to-date build environment, the system headers were different and were raising different warnings.
Rather than adding some new `#pragma`, it is cleaner to just tell the compiler that those paths are containing system headers and not user headers.